src/Health/Tests/Connection.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Tribe\Extensions\Promoter\Health\Tests;
+
+use Firebase\JWT\JWT;
+use Tribe\Extensions\Promoter\Health\Critical_Exception;
+use Tribe__Promoter__Connector;
+
+class Connection implements Runnable {
+	/**
+	 * @var Tribe__Promoter__Connector
+	 */
+	private $connector;
+	/**
+	 * @var array|\WP_Error
+	 */
+	private $response;
+	/**
+	 * @var string
+	 */
+	private $body;
+	/**
+	 * @var int
+	 */
+	private $code;
+	/**
+	 * @var string
+	 */
+	private $token = '';
+
+	public function run() {
+		$this->connector = tribe( 'promoter.connector' );
+
+		if ( ! $this->connector instanceof Tribe__Promoter__Connector ) {
+			throw new Critical_Exception( __( 'The connector class is not defined.', 'tribe_extensions_promoter' ) );
+		}
+
+		$license_info = tribe( 'promoter.pue' )->get_license_info();
+
+		if ( empty( $license_info['key'] ) ) {
+			throw new Critical_Exception( __( 'The license key of promoter is not present.', 'tribe_extensions_promoter' ) );
+		}
+
+		$payload = [
+			'licenseKey' => $license_info['key'],
+		];
+
+		$this->token = JWT::encode( $payload, get_option( 'tribe_promoter_auth_key' ) );
+		$this->request();
+
+		if ( is_wp_error( $this->response ) ) {
+			throw new Critical_Exception(
+				__( 'Error while connecting to the connector application, your site is not able to communicate with the connector site.', 'tribe_extensions_promoter' )
+			);
+		}
+
+		if ( $this->code < 200 || $this->code >= 300 ) {
+			$message = sprintf(
+				"<p>%1$s: %2$s - %3$s: %4$s</p><p>%5$s: <pre>%6$s</pre></p><p>%7$s</p>",
+				__( 'Response', 'tribe_extensions_promoter' ),
+				$this->body,
+				__( 'Code', 'tribe_extensions_promoter' ),
+				$this->code,
+				__( 'Token', 'tribe_extensions_promoter' ),
+				$this->token,
+				__( "This means that your site can't communicate correctly with WordPress, try to refresh the connection.", 'tribe_extensions_promoter' )
+			);
+
+			$actions = sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s</a></p>',
+				esc_url( 'https://promoter.theeventscalendar.com/authorization' ),
+				__( 'Authorize Again', 'tribe_extensions_promoter' )
+			);
+
+			throw new Critical_Exception( $message, $actions );
+		}
+
+		$this->validate_user();
+	}
+
+	private function request() {
+		$this->response = wp_remote_post( $this->connector->base_url() . 'connect/auth', [
+			'body'      => [ 'token' => $this->token ],
+			'timeout'   => 30,
+			'sslverify' => false,
```

Comment:
Why? Add a compelling explanation if needing to stray from the default

Reply:
https://github.com/mt-support/promoter-site-health/pull/1#discussion_r482216827